src/partial_file.rs

```diff
@@ -0,0 +1,133 @@
+use std::cmp;
+use std::fs::File;
+use iron::headers::{ByteRangeSpec, ContentLength, ContentRange, ContentRangeSpec};
+use iron::response::{WriteBody, Response};
+use iron::modifier::Modifier;
+use iron::status::Status;
+use std::io::{self, SeekFrom, Seek, Read, Write};
+use std::path::Path;
+
+pub enum PartialFileRange {
+    AllFrom(u64),
+    FromTo(u64,u64),
+    Last(u64),
+}
+
+pub struct PartialFile {
+    file: File,
+    range: PartialFileRange,
+}
+
+struct PartialContentBody {
+    pub file: File,
+    pub offset: u64,
+    pub len: u64,
+}
+
+impl PartialFile {
+    pub fn new<Range>(file: File, range: Range) -> PartialFile
+        where Range: Into<PartialFileRange> {
+        let range = range.into();
+        PartialFile {
+            file: file,
+            range: range,
+        }
+    }
+
+    /// Panics if the file doesn't exist
+    pub fn from_path<P: AsRef<Path>, Range>(path: P, range: Range) -> PartialFile
```

Comment:
The reason was originally that the file existence was already checked beforehand everytime before this method was called (via metadata). However now that I think of it this is pretty racy: if the file is deleted between the check and the file being delivered (which is unlikely but can still happen), then it will panic.

It would be better to return a `Result<_, io::Error>` there indeed.